core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -2547,6 +2547,9 @@ object SparkContext extends Logging {
   private[spark] val SPARK_SCHEDULER_POOL = "spark.scheduler.pool"
   private[spark] val RDD_SCOPE_KEY = "spark.rdd.scope"
   private[spark] val RDD_SCOPE_NO_OVERRIDE_KEY = "spark.rdd.scope.noOverride"
+  private[spark] val IS_INDETERMINATE_STAGE = "spark.stage.isIndeterminateStage"
+  private[spark] val INDETERMINATE_STAGE_ATTEMPT_ID_PREFIX =
+    "spark.stage.indeterminateStageAttemptId."
```

Comment:
can we call it `_shuffle_generation_id`? It should be internal.

Reply:
Copy that, will change it.

Reply:
Done in dc527dd.